### PR TITLE
Document AssertNextDataset op

### PR DESCRIPTION
**Pull request by frreiss**
This PR adds documentation to the API file for the `AssertNextDataset` op, which is part of `tf.data`.

### CODE REVIEW

File: tensorflow/core/api_def/base_api/api_def_AssertNextDataset.pbtxt [hunk 1]
```diff
@@ -1,4 +1,30 @@
...
+  in_arg {
+  name: "transformations"
+  description: <<END
+A `tf.string` vector `tf.Tensor` identifying the transformations that are
+expected to happen next.
+END
+  }
+  summary: "A transformation that asserts which transformations happen next."
+  description: <<END
+Core C++ implementation of the transformation `assert_next()`, which is
```
Comment (jsimsa) [inline tensorflow/core/api_def/base_api/api_def_AssertNextDataset.pbtxt]: Replace with:

```
This transformation checks whether the camel-case names (i.e. "FlatMap", not
"flat_map") of the transformations following this transformation match the list
of names in the `transformations` argument. If there is a mismatch, the
transformation raises an exception.

The check occurs when iterating over the contents of the dataset, which
means that the check happens *after* any static optimizations are applied
to the dataset graph.
```

**Review by jsimsa — COMMENTED** [review]
Thank you. Suggest minor rewording of the "description" field.

File: tensorflow/core/api_def/base_api/api_def_AssertNextDataset.pbtxt (hunk 1 above)
Comment (frreiss) [reply tensorflow/core/api_def/base_api/api_def_AssertNextDataset.pbtxt]: Thanks for the prompt review and the improvements to the description! I've pushed commit 
726351b with an updated `description` field.

File: tensorflow/core/api_def/base_api/api_def_AssertNextDataset.pbtxt [hunk 2]
```diff
@@ -1,4 +1,32 @@
...
```
Comment (jsimsa) [inline tensorflow/core/api_def/base_api/api_def_AssertNextDataset.pbtxt]: please remove this paragraph

Comment (frreiss) [reply tensorflow/core/api_def/base_api/api_def_AssertNextDataset.pbtxt]: Thanks for the follow-on review. Commit d4ab78b removes the first paragraph of the description.

**Review by jsimsa — APPROVED** [review]

**Pull request merged**